packages/mds-db/index.ts

```diff
@@ -537,6 +550,100 @@ async function readEvents(params: ReadEventsQueryParams): Promise<ReadEventsResu
   })
 }
 
+async function readHistoricalEvents(params: ReadHistoricalEventsQueryParams) {
+  const { provider_id: query_provider_id, end_time } = params
+  const client = await getReadOnlyClient()
+  const vals = new SqlVals()
+  const values = vals.values()
+  let sql = `SELECT      e2.provider_id,
```

Comment:
That's an awful lot of SQL ... Not sure I have a recommendation because views and stored procedures are also problematic, but maybe it would be easier to reason about if it was extracted into it's own function. Also, if the entire thing was constructed using template strings, you could put in new lines for readability rather than using `\n`